### PR TITLE
ci: switch to macos-14 runner

### PR DESCRIPTION
**Pull request by Dinnerbone**
Speedy!

### CODE REVIEW

File: .github/workflows/release_nightly.yml [hunk 1]
```diff
@@ -73,13 +73,13 @@ jobs:
 
           # Mac does two Rust builds to make a universal binary
           - build_name: macos-x86_64
-            os: macos-latest
+            os: macos-14
             target: x86_64-apple-darwin
```
[the remark this answers is not in the document]
Comment (Dinnerbone) [reply .github/workflows/release_nightly.yml:77]: I have no idea.

**Review by adrian17 — APPROVED** [review]

**Comment by torokati44** [comment]
Since this CI failure came up on Android as well (https://github.com/torokati44/ruffle-android/issues/63 and https://github.com/ruffle-rs/ruffle/issues/14269), here are some of my ideas to solve it:

 - Remove the NaN serialization from the test and happily forget about it being an issue (until it bites us in the rear for real sometime later).
 - Serialize all NaN values as the same fixed bit pattern - this broke some other tests IIRC.
 - Try to fiddle with the code that produces this particular NaN to make it result in the same bit pattern on both archs - seemed futile after a while...
 
My next idea was to somehow check what FP does on ARM, but I couldn't even begin to imagine how I would get my hands on an Android phone(?) with Flash still working on it...

**Comment by Dinnerbone** [comment]
> Serialize all NaN values as the same fixed bit pattern - this broke some other tests IIRC.

What other test? I think this is likely our best course.
Accept any value as NaN but write NaN as a specific value

**Comment by torokati44** [comment]
It would appear that you are, once again, right:
https://github.com/torokati44/ruffle/actions/runs/7719096570/job/21041712216#step:12:1817
https://github.com/torokati44/rust-flash-lso/commit/8a71c178426269d73001634267c688e1f7c660ba

Not cleaning this up right now, but this is roughly what you had in mind, right?
This _should_ fix the test on Android as well...

**Comment by Dinnerbone** [comment]
> Not cleaning this up right now, but this is roughly what you had in mind, right?

yep!

Things won't roundtrip, parsing something made with a different NaN turns into our fixed NaN value - but I think that's fine. We don't even know if Flash has a different NaN per arch.

**Comment by torokati44** [comment]
> What other test?

These `rust-flash-lso` tests: `user` and `user_1`... :|

**Comment by torokati44** [comment]
I'm afraid, if we're going ahead with this, we'll have to adjust those test cases, diverging from what FP produced? :/

But then, the goal of matching FP in _some other_ test cases kind of doesn't make sense anymore...?

**Comment by Dinnerbone** [comment]
I don't know :( And it's not like this is new to this PR, it's just the first time we're forced to finally confront it. We'll have to before -14 becomes -latest anyway.

Let's take this to a discord thread I guess.

**Pull request merged**